src/Symfony/Component/Form/Extension/Core/Type/ChoiceType.php

```diff
@@ -260,20 +260,19 @@ public function configureOptions(OptionsResolver $resolver)
             // Harden against NULL values (like in EntityType and ModelType)
             $choices = null !== $options['choices'] ? $options['choices'] : array();
 
-            // BC when choices are in the keys, not in the values
-            if (!$options['choices_as_values']) {
-                return $choiceListFactory->createListFromFlippedChoices($choices, $options['choice_value'], false);
-            }
-
             return $choiceListFactory->createListFromChoices($choices, $options['choice_value']);
         };
 
         $choicesAsValuesNormalizer = function (Options $options, $choicesAsValues) {
-            if (true !== $choicesAsValues) {
-                @trigger_error('The value "false" for the "choices_as_values" option is deprecated since version 2.8 and will not be supported anymore in 3.0. Set this option to "true" and flip the contents of the "choices" option instead.', E_USER_DEPRECATED);
+            if (null !== $choicesAsValues) {
+                if (true !== $choicesAsValues) {
+                    throw new \RuntimeException('The "choices_as_values" option should not be used. Remote it and flip the contents of the "choices" option instead.');
```

Comment:
I guess we would not need to do that if we didn't configure a default value so that the option wouldn't be set at all by default. But that's nitpicking and as we need to add this path anyway when deprecating the option, let's just keep the code as is.